pkg/geoipupdate/database/local_file_writer.go

```diff
@@ -0,0 +1,166 @@
+package database
+
+import (
+	"crypto/md5"
+	"fmt"
+	"github.com/gofrs/flock"
+	"github.com/pkg/errors"
+	"io"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+const zeroMD5 = "00000000000000000000000000000000"
+
+//LocalFileDatabaseWriter is a database.Writer that stores the database to the local file system
+type LocalFileDatabaseWriter struct {
+	filePath string
+	lockFile string
+	verbose  bool
+	lock     *flock.Flock
+	oldHash  string
+	swapFile *os.File
+}
+
+//NewLocalFileDatabaseWriter create a new LocalFileDatabaseWriter, creating necessary lock and swap files to protect
```

Comment:
Swap is kind of a strange term. Could we refer to it as a temporary file or something?

Reply:
Done.